test/tests/performance/pbs_preemptperformance.py

```diff
@@ -504,3 +498,102 @@ def test_preemption_basic(self):
     def tearDown(self):
         self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
         job_ids = self.server.select()
+        self.server.delete(id=job_ids)
+
+    @timeout(3600)
+    @tags('sched', 'scheduling_policy')
+    def test_insufficient_resc_multiple_non_cons2(self):
+        """
+        Submit a number of low priority jobs and then submit a high priority
+        job that needs a non-consumable resource in 2 chunks. These resources
+        are assigned to last two running jobs. This will make scheduler go
+        through all running jobs to find preemptable jobs.
+        """
+
+        a = {'type': 'string', 'flag': 'h'}
+        self.server.manager(MGR_CMD_CREATE, RSC, a, id='qlist')
+
+        a = {ATTR_rescavail + ".qlist": "list1",
+             ATTR_rescavail + ".ncpus": "8"}
+        self.server.create_vnodes(
+            "vn1", a, 1, self.mom, additive=True, fname="vnodedef1")
+
+        a = {ATTR_rescavail + ".qlist": "list2",
+             ATTR_rescavail + ".ncpus": "1"}
+        self.server.create_vnodes(
+            "vn2", a, 1, self.mom, additive=True, fname="vnodedef2")
+
+        a = {ATTR_rescavail + ".qlist": "list3",
+             ATTR_rescavail + ".ncpus": "1"}
+        self.server.create_vnodes(
+            "vn3", a, 1, self.mom, additive=True, fname="vnodedef3")
```

Comment:
If you want to create all 3 of these in one go, you can use a custom attr func.  See pbs_sched_perf.py for an example.  This does work, so it is up to you.